src/track.rs

```diff
@@ -0,0 +1,80 @@
+use itertools::Itertools;
+use log::warn;
+use serde::Serialize;
+
+#[derive(Serialize, Clone, Debug, Default)]
+/// Track
+pub struct Track {
+    pub file: String,
+    pub artist_sort: String,
+    pub album_artist: String,
+    pub album_artist_sort: String,
+    pub performer: Vec<String>,
+    pub title: String,
+    pub track: u32,
+    pub album: String,
+    pub artist: String,
+    pub pos: Option<u32>,
+    pub id: u32,
+    pub last_modified: String,
+    pub format: String,
+    pub time: u32,
+    pub duration: u32,
+    pub label: String,
+    pub date: String,
+    pub original_date: String,
+    pub disc: u32,
+    pub musicbraiz_trackid: String,
+    pub musicbrainz_albumid: String,
+    pub musicbrainz_albumartistid: String,
+    pub musicbrainz_artistid: String,
+}
+
+pub(crate) fn from_lines(lines: Vec<String>) -> Vec<Track> {
+    let mut t = Track::default();
+    let mut tracks = Vec::new();
+
+    for line in lines {
+        if let Some((k, v)) = line.splitn(2, ": ").next_tuple() {
+            match k {
+                "file" => {
```

Comment:
I think we could replace this loop with a combination of `batching()` (from itertools) and `peekable()`, i.e. batch and take as many elements until the next starts with `file: `.
Using `peekable()` gives us the ability to access the next element without consuming it.
The resulting batches could each be deserialized with serde, I think.

Reply:
I thought about that as well. We could combine those via (semi-pseudocode)
```rust
batch.sort_by(extract_key);
batch.into_iter().coalesce(|a, b| 
if a.key == b.key {
  Ok(a.key: vec![a.value, b.value])
} else {
  Err(a, b) 
})
```
using Itertools.coalesce() to group the same keys into vectors (or maybe strings like `"[a, b, c, ...]"` which could be deserialized into `Vec<T>`.

Or maybe that’s the point where I need to find out how serde parsers are written. I hope parsing this very simple format doesn’t need too much code.
But that’s definitely a topic for another day.